.cbmc-batch/stubs/memcpy_using_uint64.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * FUNCTION: memcpy
+ *
+ * This function overrides the original implementation of the memcpy function
+ * from string.h. It copies the values of n bytes from the *src to the *dst.
+ * It also checks if the size of the arrays pointed to by both the *dst and
+ * *src parameters are at least n bytes and if they overlap.]
+ *
+ * This takes advantage of the fact that 64bit operations require fewer array updates,
+ * which can make this version faster than the naive unrolling when used in CBMC.
+ * Benchmark your particular proof to know for sure.
+ */
+
+#include <stddef.h>
+#include <stdint.h>
+
+void *memcpy_using_uint64_impl(void *dst, const void *src, size_t n) {
+    __CPROVER_precondition(
+        __CPROVER_POINTER_OBJECT(dst) != __CPROVER_POINTER_OBJECT(src) ||
+            ((const char *)src >= (const char *)dst + n) || ((const char *)dst >= (const char *)src + n),
+        "memcpy src/dst overlap");
+    __CPROVER_precondition(__CPROVER_r_ok(src, n), "memcpy source region readable");
```

Comment:
don't we need to check `src  != NULL` anymore?

Reply:
This will be made redundant in CBMC https://github.com/diffblue/cbmc/pull/4485